kmir/src/kmir/cli.py

```diff
@@ -102,6 +102,12 @@ def create_argument_parser() -> ArgumentParser:
         type=dir_path,
         help='Path to Haskell definition to use.',
     )
+    prove_subparser.add_argument(
+        '--claim',
+        default=None,
+        type=str,
+        help='Provide the claim label for proving a single claim',
+    )
```

Comment:
I would make this an array, and allow the user to specify multiple claims, rather than just a single claim.

Also, we should not re-implement this option here. We should re-usie the upstream `KCLIOptions.spec_args`, and add to it: https://github.com/runtimeverification/pyk/blob/e1519523721e84eec783019f7c061121bc50241e/src/pyk/cli/args.py#L137

Here in KEVM, we re-use those arguments: https://github.com/runtimeverification/evm-semantics/blob/9e88278f72aebf74d94b928654b6364086bf165a/kevm-pyk/src/kevm_pyk/__main__.py#L736

So the idea is that pyk exposes some common functionality (`KProve.get_claims`), and then the generic options exposed via pyx's CLI module above matches that functionality directly. So the more we re-use the upstream functionality, the more things will "just work", in the sense that we don't need wrappers like `KMIR.get_all_claims` to handle the arguments.

Reply:
I am not sure why `KCLIOptions.spec_args` is not used in the first place. 
The code above was inherited from previous ones.
@dkcumming do you have any idea of why?

But if you want that way, would you please submit a seperate issue regarding the changes wanted?
It seems a much bigger changes than this PR aims at.

Reply:
From memory, I think what happened is that since mir-semantics wasn't already using KCLIArgs I just worked with the framework we already had when introducing the prover code. In retrospect I should of made an Issue so we could of discussed introducing this, but I guess I didn't think of it and was focused on the prover.

Reply:
@dkcumming I had the same impression. 
@ehildenb Well, I do not think I will make changes in this PR.
I'll suggest submit an issue in future to refactor the KCLIArgs. I would suggest maybe delay the work on it such that more requested feature could be icorperated.

Reply:
Please make such an issue befor eresolving, and link it here. See https://runtimeverification.com/blog/github-guidelines-for-collaborative-coding

Reply:
> Please make such an issue befor eresolving, and link it here. See https://runtimeverification.com/blog/github-guidelines-for-collaborative-coding

Sorry, what do you mean? I am not sure what is the expected result should. Would you please do it? 
I can link it when you made the issue. 
Thank you.

Reply:
An issue to refactor this repository to use upstream KCLIArgs class instead of making its own should be made, and then linked here.

Reply:
> An issue to refactor this repository to use upstream KCLIArgs class instead of making its entirely own CLI argument handling should be made, and then linked here.

Done https://github.com/runtimeverification/mir-semantics/issues/330